src/server/hset_family.cc

```diff
@@ -725,6 +725,23 @@ void HGetGeneric(CmdArgList args, ConnectionContext* cntx, uint8_t getall_mask)
   }
 }
 
+OpResult<vector<long>> OpHExpire(const OpArgs& op_args, string_view key, uint32_t ttl_sec,
+                                 CmdArgList values) {
+  auto& db_slice = op_args.GetDbSlice();
+  auto op_res = db_slice.FindMutable(op_args.db_cntx, key, OBJ_HASH);
+
+  if (!op_res) {
+    if (op_res.status() == OpStatus::KEY_NOTFOUND) {
+      std::vector<long> res(values.size(), -2);
+      return res;
+    }
+    return op_res.status();
+  }
```

Comment:
`If (!op_res) return op_res.status()`. The reason is because if `KEY_NOT_FOUND` we need to return an error and not an array. Also if key has `WRONG TYPE` we again need to return an error and not an array. So returning `op_res.status()` satisfies that